folder/folderutil.go

```diff
@@ -145,13 +145,30 @@ func agnosticSplit(path string) (parts []string) {
 	return
 }
 
-// HomeDirectory returns the home directory or defaultDirectory in case of error
+// HomeDirOrDefault tries to obtain the user's home directory and
+// returns the default if it cannot be obtained.
 func HomeDirOrDefault(defaultDirectory string) string {
-	usr, err := user.Current()
+	if user, err := user.Current(); err == nil && isWritable(user.HomeDir) {
+		return user.HomeDir
+	}
+	if homeDir, err := os.UserHomeDir(); err == nil && isWritable(homeDir) {
+		return homeDir
+	}
+	return defaultDirectory
+}
+
+// isWritable checks if a path is writable.
+func isWritable(path string) bool {
```

Comment:
Unless we implement a permission check helper, this function, as is, seems a good candidate to be exported